sdk/storage/azblob/service/client_test.go

```diff
@@ -711,6 +720,78 @@ func (s *ServiceUnrecordedTestsSuite) TestSASServiceClient() {
 	testcommon.ValidateBlobErrorCode(_require, err, bloberror.AuthenticationFailed)
 }
 
+func (s *ServiceUnrecordedTestsSuite) TestSASServiceClientNoKey() {
+	_require := require.New(s.T())
+	accountName := os.Getenv("AZURE_STORAGE_ACCOUNT_NAME")
+
+	serviceClient, err := service.NewClientWithNoCredential(fmt.Sprintf("https://%s.blob.core.windows.net/", accountName), nil)
+	_require.Nil(err)
+	resources := sas.AccountResourceTypes{
+		Object:    true,
+		Service:   true,
+		Container: true,
+	}
+	permissions := sas.AccountPermissions{
+		Read:                  true,
+		Write:                 true,
+		Delete:                true,
+		DeletePreviousVersion: true,
+		List:                  true,
+		Add:                   true,
+		Create:                true,
+		Update:                true,
+		Process:               true,
+		Tag:                   true,
+		FilterByTags:          true,
+		PermanentDelete:       true,
+	}
+	services := sas.AccountServices{
+		Blob:  true,
+		Queue: true,
+		File:  true,
+	}
+	start := time.Now().Add(-time.Hour)
+	expiry := start.Add(time.Hour)
+	_, err = serviceClient.GetSASURL(resources, permissions, services, expiry, &service.GetSASURLOptions{StartTime: &start})
+	_require.Error(err, "SAS can only be signed with a SharedKeyCredential")
+}
+
+func (s *ServiceUnrecordedTestsSuite) TestSASServiceClientSignNegative() {
+	_require := require.New(s.T())
+	accountName := os.Getenv("AZURE_STORAGE_ACCOUNT_NAME")
+
+	serviceClient, err := service.NewClientWithNoCredential(fmt.Sprintf("https://%s.blob.core.windows.net/", accountName), nil)
+	_require.Nil(err)
+	resources := sas.AccountResourceTypes{
+		Object:    true,
+		Service:   true,
+		Container: true,
+	}
+	permissions := sas.AccountPermissions{
+		Read:                  true,
+		Write:                 true,
+		Delete:                true,
+		DeletePreviousVersion: true,
+		List:                  true,
+		Add:                   true,
+		Create:                true,
+		Update:                true,
+		Process:               true,
+		Tag:                   true,
+		FilterByTags:          true,
+		PermanentDelete:       true,
+	}
+	services := sas.AccountServices{
+		Blob:  true,
+		Queue: true,
+		File:  true,
+	}
+	start := time.Now().Add(-time.Hour)
+	expiry := time.Time{}
+	_, err = serviceClient.GetSASURL(resources, permissions, services, expiry, &service.GetSASURLOptions{StartTime: &start})
+	_require.Error(err, "account SAS is missing at least one of these: ExpiryTime, Permissions, Service, or ResourceType")
```

Comment:
`_require.Error` checks if the error is nil or not. You will have to do `_require.Equal` if you want to match the error string.

Also the client is created using `service.NewClientWithNoCredential` here. So, it will not return the above error. You will have to create the client using shared key to get the above error.

Reply:
will fix this

Reply:
the error will return since I omitted expiry time I believe

Reply:
The first line in the `serviceClient.GetSASURL` is this,
```
if s.sharedKey() == nil {
	return "", bloberror.MissingSharedKeyCredential
}
```
Since you are creating client using `service.NewClientWithNoCredential`, you will get the `MissingSharedKeyCredential` error. You can double check this part.

Reply:
oh i see i think i misread your comment

Reply:
this should be resolved now too